src/clr-angular/utils/loading/loading.ts

```diff
@@ -7,32 +7,46 @@ import {Directive, Input, OnDestroy, Optional} from "@angular/core";
 
 import {LoadingListener} from "./loading-listener";
 
+export enum ClrLoadingState {
+    DEFAULT,
+    LOADING,
+    SUCCESS,
+    ERROR
+}
+
 @Directive({selector: "[clrLoading]"})
 export class ClrLoading implements OnDestroy {
     // We find the first parent that handles something loading
     constructor(@Optional() private listener: LoadingListener) {}
 
-    private _loading = false;
-    public get loading() {
-        return this._loading;
+    private _loadingState: ClrLoadingState = ClrLoadingState.DEFAULT;
+
+    public get loadingState() {
+        return this._loadingState;
     }
+
     @Input("clrLoading")
-    public set loading(value: boolean) {
-        value = !!value;
-        if (value === this._loading) {
+    public set loadingState(value: boolean|ClrLoadingState) {
+        if (value === this._loadingState) {
             return;
         }
-        this._loading = value;
-        if (this.listener) {
+
+        if (value === null) {
+            value = false;
+        }
+
+        if (typeof value === "boolean") {
```

Comment:
This if should go _before_ the first one, so that setting `true` when the state is already `LOADING` is ignored too.